### PR TITLE
Core/ts version task 495 market stats clean up

### PR DESCRIPTION
**Pull request by hypersolo**
## Check list  

- [ ] All Merge Request pipelines should be passed.
- [ ] Make sure all the changes are necessary and correct.
- [ ] Make sure no needed change missing (all necessary changes are made).
- [ ] Merge Request should contains only changes for one purpose (one feature or bug or refactoring).
- [ ] All necessary unit tests are written.
- [ ] Critical changes should be tested locally (or on staging).

### CODE REVIEW

File: ts-version/src/domain/entities/network.entity.ts [hunk 1]
```diff
@@ -54,10 +54,20 @@ export class Network {
...
     markets.forEach((market) => {
-      const marketStats = new MarketStats(0, 0, 0, {
-        percentage: 0,
-        absolute: 0,
-      });
+      // TODO: better initialization
+      const marketStats = new MarketStats(
+        this.name,
+        market.name,
+        {
```
Comment (WilliamHawkingStephen) [inline ts-version/src/domain/entities/network.entity.ts:61]: suggestion: 
we can use object destructuring to make it more readable and maintainable:

add option field to MarketStats constructor: 
` constructor(options: statsOptions)`
and destructure it 
` const { network, market, price, volume } = options;`

its make initialization more readable: 

`const marketStats = new MarketStats({
  network: this.name,
  market: market.name,
  price: {
    open: 0,
    high: 0,
    low: 0,
    close: 0,
    last: 0,
    change: { absolute: 0, percentage: 0 },
  },
  volume: { volume: 0, quoteVolume: 0 },
});`

File: ts-version/src/data/database/influxdb/listeners/candle.updated.ts [hunk 2]
```diff
@@ -20,20 +17,9 @@ export default class CandleUpdateListener implements IEventObserver {
...
-      low as number,
-      {} as unknown as MarketStatsChange,
-      fill.market?.name as string,
-      fill.network?.name as SupportedNetworks,
-      volume as number,
-      quoteVolume as number,
-      open as number,
-    );
-    fill.market?.setStats(stats);
+    const { volume, quoteVolume } = data[data.length - 1];
```
Comment (WilliamHawkingStephen) [inline ts-version/src/data/database/influxdb/listeners/candle.updated.ts:20]: - maybe we can use a better name definition for volume

File: ts-version/src/domain/entities/network.entity.ts (hunk 1 above)
Comment (hypersolo) [reply ts-version/src/domain/entities/network.entity.ts]: i don't think it's the best approach for constructor inputs to use json as input.
look at these lines right now, it's more readable.
thanks <3

**Review by franken5tein — APPROVED** [review]

**Review by WilliamHawkingStephen — APPROVED** [review]

File: ts-version/src/data/database/influxdb/handlers/candle.updated.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      duration: "24h",
+    };
+
+    const data = await this.candleRepo.read(input);
+    const { baseVolume, quoteVolume } = data[data.length - 1];
+    if (baseVolume !== undefined && quoteVolume !== undefined)
+      fill.market?.setVolume({ baseVolume, quoteVolume });
+  }
+
+  public static getInstance(candleRepo?: ICandleRepo.schema): IEventObserver {
```
Comment (GreatGaryCooper) [inline ts-version/src/data/database/influxdb/handlers/candle.updated.ts:23]: Can't we just make `candleRepo` parameter required and remove it from the `getInstance` check?

**Review by GreatGaryCooper — APPROVED** [review]

**Pull request merged**